public/scripts/fetch_request.js

```diff
@@ -0,0 +1,20 @@
+var fetch_request = (address, display_handler, container) => {
+
+   (function fetch_headlines(api = address){
+      fetch(api)
+         .then(function(response){
+            return response.json();
+         })
+         .then(function(myJson){
+            fetch_call_back(myJson)
+         })
+   })();
+   /* is this call back doing to much? i just liked the idea of being able to give
+      the fetche request an api address, how to display events and where - ifelt it would allow greater
+      felxibility... ?
+    */
```

Comment:
Definitely not doing too much. It's normal to have most of your app's code triggered through callbacks. What is important is that the code is neatly organised per concern.